src/shared_gui_components/OSGridController.hpp

```diff
@@ -276,6 +281,23 @@ class OSGridController : public QObject
       t_source));
   }
 
+  // This one is the canonical/correct form for a ValueType which is a ModelObject or a std::string
+  // ValueType getter() const;
+  // bool setter(const T& val);
+  // void reset();
+  // bool isDefaulted() const;
+  template <typename ValueType, typename DataSourceType>
+  void addValueEditColumn(const Heading& heading, std::function<ValueType(DataSourceType*)> getter,
+                          std::function<bool(DataSourceType*, const ValueType&)> setter,
+                          const boost::optional<std::function<void(DataSourceType*)>> reset = boost::none,
+                          const boost::optional<std::function<bool(DataSourceType*)>> isDefaulted = boost::none,
+                          const boost::optional<DataSource>& t_source = boost::none) {
+    m_baseConcepts.push_back(makeDataSourceAdapter(
+      QSharedPointer<ValueEditConcept<ValueType>>(new ValueEditConceptImpl<ValueType, DataSourceType>(heading, getter, setter, reset, isDefaulted)),
+      t_source));
+  }
+
+  // Corner case when you have a true optional with no default value
```

Comment:
new overload for const T& setter. During the https://github.com/NREL/OpenStudio/pull/4532 I made all setters that take a `std::string` take instead a `const std::string&`